docs/advanced-topics/client-anticipation.md

```diff
@@ -0,0 +1,117 @@
+
+If you need to do any sort of reanticipation to update the anticipated state of any of these variables or transforms, this method is where you will do it. This method takes as its only parameter a `double` providing the amount of time, in seconds, that the object has been rolled back (which corresponds to the round-trip time of the current batch of responses received from the server). This value can be used to calculate the difference between what the server value is, and what the anticipated client value should be, and apply that change.
+
+However, note that not all variables and transforms on that object will necessarily have received any updates, so not all of them will have necessarily been rolled back to a previous state. Before doing any reanticipation on any given variable or transform, you should check that variable/transform's `ShouldReanticipate` property. If this is `false`, then it still contains the most recent client anticipated value and no work is needed.
+
+### Global OnReanticipate
+
+In addition to the `NetworkBehaviour`'s `OnReanticipate` method, `NetworkManager` also has a callback that can be subscribed to for global reanticipation. This is useful if you need to run your reanticipation in a more global way, such as if you need to run it step-wise (say, anticipating one frame at a time) and need all objects to complete one step before any of them begin the second one. This callback receives the same `lastRoundTripTime` value as the `NetworkBehaviour` method, and is called **after** all of the `NetworkBehaviour` methods have been called.
+
+:::note
+
+If you want to implement a full client-side prediction model in your game, the global OnReanticipate callback is likely the ideal place to incorporate your rollback and replay logic. The details of implementing this, however, are left as an exercise for the user; implementing a full, production-ready prediction loop is a complex topic and recommended for advanced users only.
+
+:::
+
+## Smoothing
+
+Much of the time, when you perform an anticipation action, your anticipation will be correct and the server will update the state to the same value as you did. Occasionally, though, the server will have a different result, since latency can result in the server state changing before it receives your command in ways that affect the result. When this happens, by default, the value will be snapped to the new server value.
```

Comment:
I'd maybe try to add a link to this section at the top when explaining the basics of anticipation. I was reading the first paragraphs and I kept thinking "what about when there's desyncs?"